src/frequenz/dispatch/_bg_service.py

```diff
@@ -263,6 +264,17 @@ async def _run(self) -> None:
                         heappop(self._scheduled_events).dispatch, next_event_timer
                     )
                 elif selected_from(selected, stream):
+
+                    def is_more_relevant(
+                        dispatch: Dispatch,
+                    ) -> bool:
+                        existing_dispatch = self._dispatches.get(dispatch.id)
+
+                        return (
+                            not existing_dispatch
+                            or dispatch.update_time > existing_dispatch.update_time
+                        )
+
```

Comment:
Since you are calculating this for every case, why don't you just store it in a variable instead of creating a nested function? Also, `is_more_relevant()` seems a bit too vague, what about `is_new_or_newer`?